### ISSUE TITLE
Interpolation inside a quoted attribute value makes `isStandardSyntaxSelector` pass the whole selector over

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#170]
Split out of [#151](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/151), which teaches every check of `isStandardSyntaxSelector` to read past a quoted attribute value. One check is left doing it, because it is a util of its own and is asked elsewhere too.

### What minimal example or steps are needed to reproduce the bug?

```css
[title="#{a}"] {}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/string-quotes": "single"
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### Which Stylelint-related dependencies are you using?

```json
{
  "@stylistic/stylelint-plugin": "5.3.0",
  "stylelint": "17.14.1"
}
```

### What did you expect to happen?

`Expected single quotes`, and `--fix` writing `[title='#{a}']`.

### What actually happened?

Nothing is reported, and nothing is fixed. `hasInterpolation` finds `#{a}` in the text of the attribute value, so `isStandardSyntaxSelector` declares the selector a preprocessor construct and the rules built on it — `no-extra-semicolons`, `selector-attribute-brackets-space-inside`, `selector-descendant-combinator-no-non-space`, `selector-list-comma-newline-after`, `selector-pseudo-class-case`, `selector-pseudo-class-parentheses-space-inside`, `selector-pseudo-element-case` and `string-quotes` — pass the rule node over. `[title="@{a}"]` and `[title="$(a)"]` do the same, since the util knows all three spellings.

A `data-*` carrying a template fragment is the ordinary way to meet this: `[data-template="#{name}"]` is plain CSS, and a stylesheet holding one goes unchecked from that selector on.

### Do you have a proposal to fix the bug?

`isStandardSyntaxSelector` now blanks the quoted runs of a selector once and reads every text-matching check against the copy. `hasInterpolation` is asked before that, and on the selector itself:

```js
if (hasInterpolation(selector)) return false
```

Moving it below the blanking would answer this in one line — but `hasInterpolation` is asked by `isStandardSyntaxAtRule`, `isStandardSyntaxDeclaration`, `isStandardSyntaxProperty`, `isStandardSyntaxValue` and others, on text of their own, and each of those has the same question to answer about a string standing in it. So the change worth making is the one that settles the question in the same place for all of them, rather than a blanking in every caller.

Worth noting either way: interpolation inside a quoted value is a construct the preprocessors do read. `content: "#{$a}"` is interpolated by Sass, and so is an attribute selector's value in a `.scss` file. So the answer is not simply "a string is never syntax" — it is that a plain CSS file has no interpolation in it at all, and that the syntax being linted is what decides. `string-quotes` already asks each block of a page about its own language, per [#101](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/101), and that is the shape of the answer here as well.
